subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/tasks/AbstractNativeSourceCompileTask.java

```diff
@@ -68,4 +80,22 @@ public PreCompiledHeader getPreCompiledHeader() {
     public void setPreCompiledHeader(PreCompiledHeader preCompiledHeader) {
         this.preCompiledHeader = preCompiledHeader;
     }
+
+    /**
+     * The compiler used, including the type and the version.
+     *
+     * @since 4.4
+     */
+    @Nested
+    @Optional
+    protected VersionedNativeCompiler getCompilerIdentifier() {
+        NativeToolChainInternal toolChain = (NativeToolChainInternal) getToolChain();
+        NativePlatformInternal targetPlatform = (NativePlatformInternal) getTargetPlatform();
+        PlatformToolProvider toolProvider = toolChain.select(targetPlatform);
+        Compiler<? extends NativeCompileSpec> compiler = toolProvider.newCompiler(createCompileSpec().getClass());
+        if (!(compiler instanceof VersionedNativeCompiler)) {
+            return null;
+        }
+        return (VersionedNativeCompiler) compiler;
```

Comment:
If this is part of the public API (and it's documented, so it should be), we should only expose public types here. We could return a `NativeCompilerVersion` object, or maybe a `ToolVersion` object.

Reply:
Good catch. I created a public `NativeCompilerVersion` object which is exposed in the task.